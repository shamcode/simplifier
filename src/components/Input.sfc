<template>
    <input
        id={{id}}
        placeholder={{placeholder}}
        tabindex="{{tabindex}}"
        name={{name}}
    />
</template>

<script>
    function Input( options ) {
        options( {
            [ ref( 'placeholder' ) ]: '',
            [ ref( 'tabindex' ) ]: 1
        } );
    }

    export default Component( Template, Input );
</script>
