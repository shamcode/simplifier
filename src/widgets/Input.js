import { options } from 'sham-ui';
import Input from './Input.sht';

export default class extends Input {
    @options
    get placeholder() {
        return '';
    }

    @options
    get tabindex() {
        return 1;
    }
}