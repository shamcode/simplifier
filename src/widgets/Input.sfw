<template>
    <input
            id={{id}}
            placeholder={{placeholder}}
            tabindex="{{tabindex}}"
            name={{name}}
    />
</template>

<script>
    import { options } from 'sham-ui';

    export default class Input extends Template {
        @options placeholder = '';
        @options tabindex = 1;
    }
</script>