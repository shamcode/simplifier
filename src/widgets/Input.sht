<input
        id={{id}}
        placeholder={{placeholder}}
        tabindex="{{tabindex}}"
        name={{name}}
/>