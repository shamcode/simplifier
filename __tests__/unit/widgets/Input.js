import Input from '../../../src/widgets/Input';
import renderer from 'sham-ui-test-helpers';

it( 'renders correctly', () => {
    const meta = renderer( Input );
    expect( meta.widget.container.innerHTML ).toEqual( '<input>' );
    expect( meta.toJSON() ).toMatchSnapshot();

    meta.widget.update( {} );
    expect( meta.widget.container.innerHTML ).toEqual( '<input placeholder="" tabindex="1">' );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'with props', () => {
    const meta = renderer( Input );
    meta.widget.update( {
        id: 'expression-input',
        placeholder: 'a && b',
        tabindex: 2,
        name: 'expression'
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );



