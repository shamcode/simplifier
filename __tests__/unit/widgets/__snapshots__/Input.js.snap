// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "ID": "widget",
  "__data__": Object {},
  "html": "<input>",
  "nodes": Array [
    Object {
      "ID": undefined,
      "__data__": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
  ],
  "options": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "placeholder": "",
    "renderAsync": false,
    "renderDependence": Array [],
    "tabindex": 1,
    "types": Array [],
  },
}
`;

exports[`renders correctly 2`] = `
Object {
  "ID": "widget",
  "__data__": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "placeholder": "",
    "renderAsync": false,
    "renderDependence": Array [],
    "tabindex": 1,
    "types": Array [],
  },
  "html": "<input placeholder=\\"\\" tabindex=\\"1\\">",
  "nodes": Array [
    Object {
      "ID": undefined,
      "__data__": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
  ],
  "options": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "placeholder": "",
    "renderAsync": false,
    "renderDependence": Array [],
    "tabindex": 1,
    "types": Array [],
  },
}
`;

exports[`with props 1`] = `
Object {
  "ID": "widget",
  "__data__": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "id": "expression-input",
    "name": "expression",
    "placeholder": "a && b",
    "renderAsync": false,
    "renderDependence": Array [],
    "tabindex": 2,
    "types": Array [],
  },
  "html": "<input id=\\"expression-input\\" placeholder=\\"a &amp;&amp; b\\" tabindex=\\"2\\" name=\\"expression\\">",
  "nodes": Array [
    Object {
      "ID": undefined,
      "__data__": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
  ],
  "options": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "placeholder": "",
    "renderAsync": false,
    "renderDependence": Array [],
    "tabindex": 1,
    "types": Array [],
  },
}
`;
